src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -641,6 +655,130 @@ bool is_ip_address_assigned(void)
 	return res;
 }
 
+//
+// Format the current connection data for display to user
+//
+//	SSID = 'name'
+//	RSSI = xx
+//	ip = xxx.xxx.xxx.xxx
+//
+void wifi_get_ip_address(char *buffer, uint32_t size)
+{
+	char local_buffer[64] = {0};
+	memset(buffer, 0x00, size - 1);
+	if (g_wifi_connected) {
+		snprintf(local_buffer, sizeof(local_buffer), "SSID = %s\n", conn_info.acSSID);
+		strncpy(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "RSSI = %d\n", conn_info.s8RSSI);
+		strncat(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "IP = %d.%d.%d.%d\n", conn_info.au8IPAddr[0],
+			conn_info.au8IPAddr[1], conn_info.au8IPAddr[2], conn_info.au8IPAddr[3]);
+		strncat(buffer, local_buffer, size);
+	} else {
+		memcpy(buffer, "Not connected\n", strlen("Not connected\n"));
+	}
+}
+
+//
+// Wait for app_state to spin with timeout
+//
+void app_task_wait_spin(void)
+{
+	uint32_t wait_timeout = 2000U;
+	while (1) {
+		platform_tasks();
+		if (app_state == app_state_spin)
+			break;
+		platform_delay(1);
+		if (wait_timeout-- == 0)
+			break;
+	}
+}
+
+//
+// Using the passed arguments, attempt to connect to a Wi-Fi AP
+//
+void wifi_connect(int argc, const char **argv, char *buffer, uint32_t size)
+{
+	char ssid[64] = {0};
+	char pass_phrase[64] = {0};
+	char *output_buffer = ssid;
+	char *delimeter;
+	bool add_space = false;
+	uint32_t transfer_count = 0;
+	memset(buffer, 0x00, size);
+	//
+	// Iterate over the arguments received to build the SSID and passphrase
+	//
+	// The BMF command line parser treats spaces as delimiters and both SSID
+	// and the passphrase may have embedded spaces. The SSID and passphrase
+	// should have a comma separator.
+	//
+	// The following loop concatenates each arg and adds back the space delimiter.
+	//
+	// It does this initially into the SSID name, up until an argument is found
+	// that contains a comma. The string before the comma is concatenated with
+	// the SSID and the string after the comma is used as the first element of
+	// the passphrase.
+	//
+	// The remaining arguments are then concatenated into the passphrase with
+	// an space added between them.
+	//
+	for (int loop = 1; loop < argc; loop++) {
+		delimeter = strchr(argv[loop], ',');
+		if (delimeter == NULL) {
+			if (add_space) {
+				strcat(output_buffer, " ");
+				add_space = true;
```

Comment:
This looks to be wrong logic? Surely this should be assigning `false` to this variable having performed the action the variable signals (to reset for next loop).

Reply:
INdeed, not only the wrong logic, also this parser did not handle multiple spaces in the arguments to the "wifi" monitor command.